OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -111,10 +111,10 @@ public override bool HandleMouseInput(MouseInput mi)
 
 						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
 						{
-							var newSelection2 = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
-								a => a.Owner == unit.Owner && a.Info.Name == unit.Info.Name);
+							var newSelection = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
+								a => a.Owner == unit.Owner && a.Trait<Selectable>().Class == unit.Trait<Selectable>().Class);
```

Comment:
Repeated trait lookups is not the best thing to do, so add a `var class = unit.Trait<Selectable>().Class;` above and check against that here.